--- a/etc/rc.start_packages	2024-12-11 13:07:08
+++ b/etc/rc.start_packages	2024-12-11 13:10:08
@@ -61,10 +61,29 @@
 	}
 }
 
+$frrdelay = false;
+
 foreach (config_get_path('installedpackages/package', []) as $pkgid => $package) {
-	echo " Starting package {$package['name']}...";
-	sync_package($package['name']);
-	$internal_name = get_package_internal_name($package);
+	if ($package['name'] == 'FRR') {
+		echo " Skipping FRR for now...";
+		$frrdelay = true;
+		continue;
+	} else{
+		echo " Starting package {$package['name']}...";
+		sync_package($package['name']);
+		$internal_name = get_package_internal_name($package);
+		start_service($internal_name, true);
+		unset($rcfiles[RCFILEPREFIX . strtolower($internal_name) . ".sh"]);
+		echo "done.\n";
+	}
+}
+
+if ($frrdelay) {
+	echo " Delaying 30 seconds to allow Wireguard to reconfigure interfaces...";
+	sleep(30);
+	echo " Starting package FRR...";
+	sync_package('FRR');
+	$internal_name = 'frr';
 	start_service($internal_name, true);
 	unset($rcfiles[RCFILEPREFIX . strtolower($internal_name) . ".sh"]);
 	echo "done.\n";
  
